### PR TITLE
Improve Documentation VI (Docstrings + Asserts)

### PR DESCRIPTION
**Pull request by zm711**
Hey all, with the change to main->main I wanted to do a small test before I do any bigger sweeps of docstrings, asserts, errors etc for generating the API documentation. Still just fixing minor typos and trying to clarify a few minor points-no code changes.

### CODE REVIEW

File: src/spikeinterface/curation/curationsorting.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class CurationSorting:
         passed to the new units (if the original units have the same value). If 'remove' the new units will have 
         an empty value for all the properties. Default: 'keep'
     make_graph: bool
-        True to keep a networkx graph with the curation history
+        True to keep a network graph with the curation history
```
Comment (h-mayorquin) [inline src/spikeinterface/curation/curationsorting.py:20]: I think networkx was the intention 
```suggestion
        True to keep a Networkx graph instance with the curation history
```

File: src/spikeinterface/curation/mergeunitssorting.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 class MergeUnitsSorting(BaseSorting):
     """
-    Class that handles several merges of units from a Sorting object based on a list of list of unit_ids.
+    Class that handles several merges of units from a Sorting object based on a list of lists of unit_ids.
```
Comment (h-mayorquin) [inline src/spikeinterface/curation/mergeunitssorting.py:8]: I still find this confusing
```suggestion
    Class that handles several merges of units from a Sorting object based on a list whose elements are lists of unit_ids.
```
What do you think?

File: src/spikeinterface/curation/remove_duplicated_spikes.py [hunk 3]
```diff
@@ -24,8 +24,8 @@ class RemoveDuplicatedSpikesSorting(BaseSorting):
         If method = "keep_last", for each ISI violation, will remove the first spike.
         If method = "keep_first_iterative", will iteratively keep the first spike and remove the following violations.
         If method = "keep_last_iterative", does the same as "keep_first_iterative" but starting from the end.
-        In the iterative methods, if there is a triplet A, B, C where (A, B) and (A, C) are in the censored period
-        (but not (A, C)), then only B is removed. In the non iterative method however, only one spike remains.
+        In the iterative methods, if there is a triplet A, B, C where (A, B) and (B, C) are in the censored period
```
Comment (h-mayorquin) [inline src/spikeinterface/curation/remove_duplicated_spikes.py:27]: Thanks for this one!

File: src/spikeinterface/curation/remove_excess_spikes.py [hunk 4]
```diff
@@ -60,14 +60,14 @@ def get_unit_spike_train(self, unit_id, start_frame: Optional[int] = None,
...
-    Excess spikes are the ones exceeding a recording number of samples, for each segment
+    Excess spikes are the ones exceeding a recording number of samples, for each segment.
 
     Parameters
     ----------
     sorting: BaseSorting
         The parent sorting.
     recording: BaseRecording
-        The recording to use to get number of samples.
+        The recording to use to get the number of samples.
```
Comment (h-mayorquin) [inline src/spikeinterface/curation/remove_excess_spikes.py:70]:
```suggestion
        The recording used to get the number of samples.
```
?

**Review by h-mayorquin — APPROVED** [review]
This looks good to me, I added some possible suggestions. 

Sorry for the recent changes and thanks for this PR. Now we have a stable main that is actually the default branch right now. We won't be making such big movement of files again.

File: src/spikeinterface/curation/curationsorting.py (hunk 1 above)
Comment (alejoe91) [reply src/spikeinterface/curation/curationsorting.py:20]: Indeed! Good cathc @h-mayorquin

File: src/spikeinterface/curation/remove_excess_spikes.py (hunk 4 above)
Comment (alejoe91) [reply src/spikeinterface/curation/remove_excess_spikes.py:70]: I think it is fine as is

File: src/spikeinterface/curation/mergeunitssorting.py (hunk 2 above)
Comment (alejoe91) [reply src/spikeinterface/curation/mergeunitssorting.py:8]: I think list of lists is clear enough :)

**Pull request merged**